gqlalchemy/query_builders/memgraph_query_builder.py

```diff
@@ -79,8 +80,109 @@ def load_csv(self, path: str, header: bool, row: str) -> "DeclarativeBase":
 
         return self
 
+    def construct_subgraph_query(
+        self, node_label: Optional[str] = None, relationship_types: Optional[Union[str, Iterable[str]]] = None
+    ) -> str:
+        """Constructs a MATCH query defining a subgraph using zero or one node labels and relationship types.
+
+        Args:
+            node_label (Optional[str], optional): node label to be used in subgraph
+            relationship_types (Optional[Union[str, Iterable[str]]], optional): types of relationships used in subgraph.
+
+        Returns:
+            str: a MATCH query for a path with given node labels and relationship types
+        """
+        query = " MATCH p="
+        if node_label is None:
+            node = "()"
+        else:
+            node = f"(:{node_label.upper()})"
+
+        query += node + "-"
+
+        if isinstance(relationship_types, str):
+            query += f"[:{relationship_types}]"
+        elif isinstance(relationship_types, Iterable):
+            rels = " | :".join(relationship_types)
+            query += f"[:{rels}]"
+
+        # directed is OK only if we have single label, homogenous graphs
+        query += "->" + node
```

Comment:
One more thing also poped into my mind - we can create paths in that way. A path is node-relationship-node. If we have these defined, we just append them. That can be used to simplify your code below.